Weather.cpp reference 'dateTimeController' is not yet bound

### Verification

- [X] I searched for similar bug reports and found none was relevant.

### What happened?

Clang issues a warning when compiling

### What should happen instead?

_No response_

### Reproduction steps

compile the project

### More details?

```
InfiniTime/src/displayapp/screens/Weather.cpp:30:25: warning: reference 'dateTimeController' is not yet bound to a value when used here [-Wuninitialized]
    dateTimeController {dateTimeController},
                        ^
1 warning generated.
```
@Avamander git blames says the line in question was written by you

It's just a debugging tool, but wanted to inform you about the usage of a reference, which isn't initialized

### Version

21da5869c5e48df666ea9d9c7698f7f69528645e

### Companion app

_No response_

--- a/src/displayapp/screens/Weather.cpp
+++ b/src/displayapp/screens/Weather.cpp
@@ -27,7 +27,6 @@ using namespace Pinetime::Applications::Screens;
 
 Weather::Weather(Pinetime::Applications::DisplayApp* app, Pinetime::Controllers::WeatherService& weather)
   : Screen(app),
-    dateTimeController {dateTimeController},
     weatherService(weather),
     screens {app,
              0,

--- a/src/displayapp/screens/Weather.h
+++ b/src/displayapp/screens/Weather.h
@@ -25,7 +25,6 @@ namespace Pinetime {
       private:
         bool running = true;
 
-        Pinetime::Controllers::DateTime& dateTimeController;
         Controllers::WeatherService& weatherService;
 
         ScreenList<5> screens;

--- a/src/displayapp/widgets/Counter.cpp
+++ b/src/displayapp/widgets/Counter.cpp
@@ -28,7 +28,7 @@ namespace {
   }
 }
 
-Counter::Counter(int min, int max, lv_font_t& font) : min {min}, max {max}, value {min}, font {font}, leadingZeroCount {digitCount(max)} {
+Counter::Counter(int min, int max, lv_font_t& font) : min {min}, max {max}, value {min}, leadingZeroCount {digitCount(max)}, font {font} {
 }
 
 void Counter::UpBtnPressed() {

